packages/amplify-category-function/src/provider-utils/awscloudformation/utils/packageLayer.ts

```diff
@@ -88,17 +96,30 @@ async function ensureLayerVersion(context: any, layerName: string) {
   }
   await layerState.setNewVersionHash(); // "finialize" the latest layer version
   const storedParams = layerState.toStoredLayerParameters();
-  const additionalLayerParams = {
+  const additionalLayerParams: {
+    layerName: string;
+    build: boolean;
+    providerContext: {
+      provider: string;
+      service: string;
+      projectName: string;
+    };
+    runtimes?: LayerRuntime[];
+  } = {
     layerName,
     build: true,
     providerContext: {
       provider,
       service: ServiceName.LambdaLayer,
       projectName: context.amplify.getProjectDetails().projectConfig.projectName,
     },
-    runtimes: getLayerRuntimes(context.amplify.pathManager.getBackendDirPath(), layerName),
   };
-  const layerParameters: LayerParameters = { ...storedParams, ...additionalLayerParams };
+
+  if (isMultiEnvLayer(context, layerName)) {
+    additionalLayerParams.runtimes = getLayerRuntimes(context.amplify.pathManager.getBackendDirPath(), layerName);
```

Comment:
not completely necessary, but using the new pathManager from amplify-cli-core here and above would be preferable